packages/pyright-internal/src/analyzer/sourceFile.ts

```diff
@@ -115,6 +115,13 @@ export const getIPythonCells = (fileSystem: FileSystem, uri: Uri, console: Conso
     }
     try {
         const parsedNotebook = JSON.parse(fileContent) as INotebookContent;
+
+        // https://github.com/microsoft/vscode/blob/7e4e0f4e55d0d0a2a931aa4e0b7acc518e5da0dd/extensions/ipynb/src/notebookSerializer.ts#L49
+        // VSCode does not support jupyter nbformat 3 since 1.63.0
+        if (parsedNotebook.nbformat && parsedNotebook.nbformat < 4) {
+            return;
```

Comment:
this causes pyright to assume it's a regular python file, which also isn't correct. i think we should just throw an error here instead
```suggestion
            throw new Error(`only nbformat 4 or above is supported (received ${parsedNotebook.nbformat})`);
```

Reply:
I think we should treat it as a plain text file. There shouldn't be any errors displayed in the UI. When a user opens a workspace, our system automatically starts the indexing process. And users might suddenly see error messages in the UI even when they haven't performed any operations on the files yet. Also, it seems that Pylance isn't throwing any errors either.

Reply:
And user cannot open this notebook because VSCode is refused to render this file:
<img width="785" alt="image" src="https://github.com/user-attachments/assets/e33314cb-98fd-4fb3-b49f-c832fbcc7c39" />

Reply:
the issue with this solution though is that it only works in the language server. the CLI still incorrectly treats it as a regular python file and attempts to parse the json as python code, which results in all these false positives in your example file:

```
/home/me/basedpyright/basedpyright/asdf.ipynb
  /home/me/basedpyright/basedpyright/asdf.ipynb:1:1 - warning: Expression value is unused (reportUnusedExpression)
  /home/me/basedpyright/basedpyright/asdf.ipynb:36:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:62:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:74:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:110:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:123:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:160:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:198:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:225:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:256:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:283:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:302:34 - error: "false" is not defined (reportUndefinedVariable)
  /home/me/basedpyright/basedpyright/asdf.ipynb:336:34 - error: "false" is not defined (reportUndefinedVariable)
12 errors, 1 warning, 0 notes
```

while this is the same behavior that pyright has on all notebooks, i don't think we should fall back to it because it's very confusing to the user, especially because unlike pyright we explicitly state that we support notebooks.

Reply:
thanks for understanding, in the mean time i've opened #1341 to track this issue